x/upgrade/abci_test.go

```diff
@@ -185,6 +191,35 @@ func (s *TestSuite) TestPlanStringer() {
   Info: `, Plan{Name: "test", Height: 100}.String())
 }
 
+func (s *TestSuite) TestSkipUpgrade()  {
+	newCtx := sdk.NewContext(s.cms, abci.Header{Height: s.ctx.BlockHeight() + 1, Time: time.Now()}, false, log.NewNopLogger())
+	req := abci.RequestBeginBlock{Header: newCtx.BlockHeader()}
+	err := s.handler(s.ctx, SoftwareUpgradeProposal{Title: "prop", Plan: Plan{Name: "test", Height: s.ctx.BlockHeight() + 1}})
+	s.Require().Nil(err)
+
+	s.T().Log("Verify if skip upgrade flag clears upgrade plan")
+	viper.Set(s.FlagUnsafeSkipUpgrade, true )
+	s.Require().NotPanics(func() {
+		s.module.BeginBlock(newCtx, req)
+	})
+	s.VerifyCleared(s.ctx)
```

Comment:
Please also add a check that this was not actually executed - that the names plan is not listed in done either

Reply:
Yeah,missed that.Will change.